src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/EntryManagerTest.php

```diff
@@ -44,4 +44,37 @@ public function testGetResources()
         $entryManager = new EntryManager($connection);
         $entryManager->update($entry);
     }
+
+    /**
+     * @see https://tools.ietf.org/html/rfc4514#section-3
+     *
+     * @dataProvider moveWithRFC4514DistinguishedNameProvider
+     */
+    public function testMoveWithRFC4514DistinguishedName(string $dn, string $expectedRdn)
+    {
+        $connection = $this->createMock(Connection::class);
+
+        $entry = new Entry($dn);
+        $entryManager = new EntryManager($connection);
+
+        $method = (new \ReflectionClass(EntryManager::class))->getMethod('parseRdnFromEntry');
+        $method->setAccessible(true);
+
+        $cn = $method->invokeArgs($entryManager, [$entry, 'a']);
+
+        $this->assertSame($expectedRdn, $cn);
+    }
+
+    /**
+     * @return array<array{0: string, 1: string}>
+     */
+    public function moveWithRFC4514DistinguishedNameProvider(): array
+    {
+        return [
+            ['CN=Simple,DC=example,DC=net', 'CN=Simple'],
+            ['CN=James \"Jim\" Smith\, III,DC=example,DC=net', 'CN=James \"Jim\" Smith\, III'],
```

Comment:
Sorry if it's a dumb question: should the expected value be `'CN=James "Jim" Smith, III'`?

Reply:
Certain characters in the canonical name must be escaped for them to work with LDAP server. 

So to continue working with the LDAP server we need the backslash in front of the quotes.  => **'CN=James \"Jim\" Smith\, III'**